src/test/java/com/easypost/ShipmentTest.java

```diff
@@ -0,0 +1,224 @@
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testBuy() throws EasyPostException {
+        Shipment shipment = Shipment.create(Fixture.basicShipment());
+
+        Shipment boughtShipment = shipment.buy(shipment.lowestRate());
+
+        assertTrue(boughtShipment instanceof Shipment);
+        assertNotNull(boughtShipment.getPostageLabel());;
+    }
+
+    /**
+     * Test generating a new rate for a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRegenerateRates() throws EasyPostException {
+        Shipment shipmentWithNewRates = shipment.newRates();
+
+        List<Rate> rates = shipmentWithNewRates.getRates();
+
+        assertTrue(rates instanceof List);
+        for(Rate rate: rates) {
+            assertTrue(rate instanceof Rate);
+        }
+    }
+
+    /**
+     * Test converting a label for a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testConvertLabel() throws EasyPostException {
+        Map<String, Object> params = new HashMap<>();
+
+        params.put("file_format", "ZPL");
+
+        Shipment shipment = Shipment.create(Fixture.oneCallBuyShipment());
+        
+        Shipment shipmentWithLabel = shipment.label(params);
+
+        assertTrue(shipmentWithLabel instanceof Shipment);
+        assertNotNull(shipmentWithLabel.getPostageLabel().getLabelZplUrl());
+    }
+
+    /**
+     * Test adding insurace to a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testInsure() throws EasyPostException {
+        Map<String, Object> shipmentData = Fixture.oneCallBuyShipment();
+        Map<String, Object> insuranceData = new HashMap<>();
+
+        insuranceData.put("amount", "100");
+        shipmentData.put("insurance", 0);
+
+        Shipment shipment = Shipment.create(shipmentData);
+
+        Shipment shipmentWithInsurance = shipment.insure(insuranceData);
+
+        assertTrue(shipmentWithInsurance instanceof Shipment);
+        assertEquals("100.00", shipmentWithInsurance.getInsurance());
+    }
+
+    /**
+     * Test refunding a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testRefund() throws EasyPostException {
+        Shipment shipment = Shipment.create(Fixture.oneCallBuyShipment());
+
+        Shipment refundedShipment = shipment.refund();
+
+        assertTrue(refundedShipment instanceof Shipment);
+        assertEquals("submitted", refundedShipment.getRefundStatus());
+    }
+
+    /**
+     * Test getting smart rate from a shipment.
+     *
+     * @throws EasyPostException when the request fails.
+     */
+    @Test
+    public void testSmartRate() throws EasyPostException {
+        List<Rate> smartRates = shipment.getSmartrates();
+
+        assertTrue(smartRates instanceof List);
```

Comment:
Please write this test like it's written in the other libraries. Asserting that each field is filled. We wrote it this way previously because we had initial issues on those fields being null.